enterprise/frontend/src/embedding-sdk/test/sdk-compatibility-warning.unit.spec.tsx

```diff
@@ -0,0 +1,116 @@
+import { render } from "@testing-library/react";
+import fetchMock from "fetch-mock";
+
+import {
+  setupCurrentUserEndpoint,
+  setupPropertiesEndpoints,
+} from "__support__/server-mocks";
+import { waitForLoaderToBeRemoved } from "__support__/ui";
+import {
+  MetabaseProvider,
+  defineEmbeddingSdkConfig,
+} from "embedding-sdk/components/public";
+import {
+  createMockSettings,
+  createMockTokenFeatures,
+  createMockUser,
+  createMockVersion,
+} from "metabase-types/api/mocks";
+
+import { getEmbeddingSdkVersion } from "../config";
+
+// TODO: extract this common setup to a shared util
+const METABASE_INSTANCE_URL = "path:";
+const AUTH_PROVIDER_URL = "http://auth-provider/metabase-sso";
+
+const defaultAuthUriConfig = defineEmbeddingSdkConfig({
+  metabaseInstanceUrl: METABASE_INSTANCE_URL,
+  authProviderUri: AUTH_PROVIDER_URL,
+  fetchRequestToken: _ =>
+    Promise.resolve({
+      id: "123",
+      exp: Number.MAX_SAFE_INTEGER,
+    }),
+});
+
+jest.mock("../config", () => ({
+  getEmbeddingSdkVersion: jest.fn(),
+}));
+
+const setup = async ({
+  sdkVersion,
+  mbVersion,
+}: {
+  sdkVersion: string;
+  mbVersion: string;
+}) => {
+  (getEmbeddingSdkVersion as jest.Mock).mockReturnValue(sdkVersion);
+  setupPropertiesEndpoints(
+    createMockSettings({
+      "token-features": createMockTokenFeatures({
+        embedding_sdk: true,
+      }),
+      version: createMockVersion({ tag: mbVersion }),
+    }),
+  );
+  setupCurrentUserEndpoint(createMockUser({ id: 1 }));
+
+  render(
+    <MetabaseProvider config={defaultAuthUriConfig}>
+      <div>Hello</div>
+    </MetabaseProvider>,
+  );
+  await waitForLoaderToBeRemoved();
+};
+
+let consoleWarnSpy: jest.SpyInstance;
+
+const getWarnMessages = (): string[] =>
+  consoleWarnSpy.mock.calls.map(call => call.join(" "));
```

Comment:
`call` as an array confuses me a bit, is it more like `callArguments` or something? I'm not sure.

I know `mock.calls` shape is a bit weird and I don't ever remember how it looks like.

Reply:
Yeah it's a bit confusing, but in the end what they store for a "function call" is the args so I understand why they did it, I still find it confusing, I'll rename as suggested